### ISSUE TITLE
[Bug]: [standalone] Panic when graceful exit before init succeeded

### ISSUE THREAD

**Issue by haorenfsa** [milvus-io/milvus#14165]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.0.0-testing-20211222-50-g1a80e65-dev
- Deployment mode(standalone or cluster): standalone
- OS(Ubuntu or CentOS): Centos
```

### Current Behavior

panic when graceful exit before init finished

log:
```
12月 24 13:51:54 knova1 systemd[1]: Started Milvus Standalone Server.
12月 24 13:51:54 knova1 systemd[1]: Starting Milvus Standalone Server...
12月 24 13:51:54 knova1 milvus-server[11070]: __  _________ _   ____  ______
12月 24 13:51:54 knova1 milvus-server[11070]: /  |/  /  _/ /| | / / / / / __/
12月 24 13:51:54 knova1 milvus-server[11070]: / /|_/ // // /_| |/ / /_/ /\ \
12月 24 13:51:54 knova1 milvus-server[11070]: /_/  /_/___/____/___/\____/___/
12月 24 13:51:54 knova1 milvus-server[11070]: Welcome to use Milvus!
12月 24 13:51:54 knova1 milvus-server[11070]: Version:   v2.0.0-testing-20211222-50-g1a80e65-dev
12月 24 13:51:54 knova1 milvus-server[11070]: Built:     Fri Dec 24 03:45:07 UTC 2021
12月 24 13:51:54 knova1 milvus-server[11070]: GitCommit: 1a80e65
12月 24 13:51:54 knova1 milvus-server[11070]: GoVersion: go version go1.15.2 linux/amd64
12月 24 13:51:54 knova1 milvus-server[11070]: open pid file: /run/milvus/standalone.pid
12月 24 13:51:54 knova1 milvus-server[11070]: lock pid file: /run/milvus/standalone.pid
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.776 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.778 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.781 +08:00] [DEBUG] [global_rmq.go:50] ["initializing global rmq"] [path=/var/lib/milvus/rdb_data]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.781 +08:00] [DEBUG] [global_rmq.go:93] [RocksmqRetentionTimeInMinutes=10080] [RocksmqRetentionSizeInMB=8192] [RocksmqPageSize=2147483648
12月 24 13:51:54 knova1 milvus-server[11070]: 2021/12/24 13:51:54 sync and save timestamp{last 16 1640325122742776218  Local} {save 16 1640325125743776218  Local} {next 16 1640325122743776218  Local}
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.873 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.875 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.876 +08:00] [DEBUG] [global_param.go:139] [initServerMaxSendSize] [role=rootcoord] [grpc.serverMaxSendSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.876 +08:00] [DEBUG] [global_param.go:162] [initServerMaxRecvSize] [role=rootcoord] [grpc.serverMaxRecvSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.876 +08:00] [DEBUG] [service.go:152] ["grpc init done ..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.877 +08:00] [DEBUG] [service.go:157] ["init params done"]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.877 +08:00] [DEBUG] [service.go:229] ["start grpc "] [port=53100]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.877 +08:00] [DEBUG] [rocksmq_retention.go:87] ["Rocksmq retention goroutine start!"]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.877 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.879 +08:00] [DEBUG] [proxy.go:107] ["create a new Proxy instance"] [module=Proxy] [state=Abnormal]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.879 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.880 +08:00] [DEBUG] [global_param.go:139] [initServerMaxSendSize] [role=proxy] [grpc.serverMaxSendSize=536870912]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.880 +08:00] [DEBUG] [global_param.go:162] [initServerMaxRecvSize] [role=proxy] [grpc.serverMaxRecvSize=536870912]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.880 +08:00] [DEBUG] [service.go:156] ["init params done ..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.880 +08:00] [DEBUG] [service.go:167] [proxy] ["proxy host"=172.19.164.152]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.880 +08:00] [DEBUG] [service.go:168] [proxy] ["proxy port"=19530]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.880 +08:00] [DEBUG] [service.go:169] [proxy] ["proxy address"=172.19.164.152:19530]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.880 +08:00] [DEBUG] [service.go:106] [proxy] ["network port"=19530]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.881 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.882 +08:00] [DEBUG] [query_coord.go:279] ["query coordinator"] [queryChannels="[{}]"]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.883 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.885 +08:00] [DEBUG] [global_param.go:139] [initServerMaxSendSize] [role=querycoord] [grpc.serverMaxSendSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.885 +08:00] [DEBUG] [global_param.go:162] [initServerMaxRecvSize] [role=querycoord] [grpc.serverMaxRecvSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.889 +08:00] [DEBUG] [service.go:232] [network] [port=19531]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.890 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.893 +08:00] [DEBUG] [param_table.go:176] ["init cacheSize"] ["cacheSize (GB)"=0]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.893 +08:00] [DEBUG] [param_table.go:327] ["query node init gracefulTime"] [gracefulTime=0]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.893 +08:00] [DEBUG] [param_table.go:337] ["initialize the knowhere simd type"] [simd_type=auto]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.893 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.896 +08:00] [DEBUG] [global_param.go:139] [initServerMaxSendSize] [role=querynode] [grpc.serverMaxSendSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.896 +08:00] [DEBUG] [global_param.go:162] [initServerMaxRecvSize] [role=querynode] [grpc.serverMaxRecvSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.896 +08:00] [DEBUG] [service.go:95] [QueryNode] [port=21123]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.897 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.898 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.900 +08:00] [DEBUG] [global_param.go:139] [initServerMaxSendSize] [role=datacoord] [grpc.serverMaxSendSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.900 +08:00] [DEBUG] [global_param.go:162] [initServerMaxRecvSize] [role=datacoord] [grpc.serverMaxRecvSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.900 +08:00] [DEBUG] [service.go:116] ["network port"] [port=13333]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.901 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.903 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.904 +08:00] [DEBUG] [global_param.go:139] [initServerMaxSendSize] [role=datanode] [grpc.serverMaxSendSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.904 +08:00] [DEBUG] [global_param.go:162] [initServerMaxRecvSize] [role=datanode] [grpc.serverMaxRecvSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.904 +08:00] [DEBUG] [service.go:203] ["DataNode address"] [address=172.19.164.152:21124]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.904 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.909 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.911 +08:00] [DEBUG] [global_param.go:139] [initServerMaxSendSize] [role=indexcoord] [grpc.serverMaxSendSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.911 +08:00] [DEBUG] [global_param.go:162] [initServerMaxRecvSize] [role=indexcoord] [grpc.serverMaxRecvSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.913 +08:00] [DEBUG] [service.go:199] [IndexCoord] ["network address"=172.19.164.152] ["network port"=31000]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.913 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.915 +08:00] [DEBUG] [param_table.go:174] ["initialize the knowhere simd type"] [simd_type=auto]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.915 +08:00] [DEBUG] [indexnode.go:97] ["New IndexNode ..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.915 +08:00] [DEBUG] [base_table.go:60] ["config directory"] [configDir=/etc/milvus/configs/]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.918 +08:00] [DEBUG] [global_param.go:139] [initServerMaxSendSize] [role=indexnode] [grpc.serverMaxSendSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.918 +08:00] [DEBUG] [global_param.go:162] [initServerMaxRecvSize] [role=indexnode] [grpc.serverMaxRecvSize=2147483647]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.919 +08:00] [DEBUG] [service.go:78] [IndexNode] ["network address"=172.19.164.152:21121] ["network port: "=21121]
12月 24 13:51:54 knova1 systemd[1]: Stopping Milvus Standalone Server...
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [ERROR] [roles.go:462] ["Get signal to exit\n"] [signal=terminated] [stack="github.com/milvus-io/milvus/cmd/roles.(*MilvusRo
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:171] ["IndexNode stop"] [Address=172.19.164.152:21121]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [indexnode.go:244] ["Index node stopped."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:182] ["Graceful stop grpc server..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:119] ["Indexcoord stop"] [Address=172.19.164.152:31000]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:157] ["Datanode stop"] [Address=172.19.164.152:21124]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:165] ["Graceful stop grpc server..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:171] ["Datacoord stop"] [Address=172.19.164.152:13333]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:181] ["Graceful stop grpc server..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:257] ["QueryNode stop"] [Address=172.19.164.152:21123]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:266] ["Graceful stop grpc server..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:270] ["QueryCoord stop"] [Address=172.19.164.152:19531]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:271] ["Proxy stop"] [Address=172.19.164.152:19530]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:280] ["Graceful stop grpc server..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:270] ["Rootcoord stop"] [Address=172.19.164.152:53100]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:296] ["Rootcoord begin to stop grpc server"]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [service.go:299] ["Graceful stop grpc server..."]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [global_rmq.go:102] ["Close Rocksmq!"]
12月 24 13:51:54 knova1 milvus-server[11070]: [2021/12/24 13:51:54.933 +08:00] [DEBUG] [rocksmq_retention.go:95] ["Rocksmq retention finish!"]
12月 24 13:51:54 knova1 milvus-server[11070]: panic: runtime error: invalid memory address or nil pointer dereference
12月 24 13:51:54 knova1 milvus-server[11070]: panic: runtime error: invalid memory address or nil pointer dereference
12月 24 13:51:54 knova1 milvus-server[11070]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x18216a2]
12月 24 13:51:54 knova1 milvus-server[11070]: goroutine 1 [running]:
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).Close(0x0)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/internal/querycoord/task_scheduler.go:864 +0x22
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/internal/querycoord.(*QueryCoord).Stop(0xc00021f700, 0x0, 0x0)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/internal/querycoord/query_coord.go:239 +0x6c
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).Stop(0xc0005c2e60, 0xc0004c5588, 0x0)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/internal/distributed/querycoord/service.go:276 +0x1fe
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/cmd/components.(*QueryCoord).Stop(0xc00064a200, 0xc000044000, 0xc0004dfae8)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/cmd/components/query_coord.go:59 +0x2f
12月 24 13:51:54 knova1 milvus-server[11070]: panic(0x2cbfbc0, 0x40d4020)
12月 24 13:51:54 knova1 milvus-server[11070]: /usr/local/go/src/runtime/panic.go:975 +0x3e9
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/internal/indexcoord.(*TaskScheduler).Close(0x0)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/internal/indexcoord/task_scheduler.go:302 +0x22
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).Stop(0xc000291e00, 0x0, 0x0)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/internal/indexcoord/index_coord.go:296 +0x74
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).Stop(0xc0004006c0, 0xc0004c58d8, 0xc0004c57e8)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/internal/distributed/indexcoord/service.go:126 +0x27d
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/cmd/components.(*IndexCoord).Stop(0xc0003e9088, 0xc0004df978, 0x7e39d6)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/cmd/components/index_coord.go:57 +0x2e
12月 24 13:51:54 knova1 milvus-server[11070]: github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc0004c5ef0, 0x1, 0x0, 0x0)
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/cmd/roles/roles.go:466 +0x5b3
12月 24 13:51:54 knova1 milvus-server[11070]: main.main()
12月 24 13:51:54 knova1 milvus-server[11070]: /data/milvus/cmd/main.go:294 +0x635
```

### Expected Behavior

not panic

### Steps To Reproduce

```markdown
1. `milvus run standalone`
2. immediately `killall milvus`
```

**Comment by yanliang567** [comment]
/assign @xiaofan-luan 
@xiaofan-luan is it something you are looking at ?

/unassign

**Comment by haorenfsa** [comment]
@DragonDriver still panics, I add some logs might help, FYI:

code:
![image](https://user-images.githubusercontent.com/15938850/147727650-4e742c3c-d2a4-4aec-9eec-fb2d05e8a0ad.png)

log:
``` log
^C[2021/12/30 06:33:44.261 +00:00] [ERROR] [roles.go:496] ["Get signal to exit\n"] [signal=interrupt] [stack="github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run\n\t/data/milvus/cmd/roles/roles.go:496\nmain.main\n\t/data/milvus/cmd/main.go:296\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:204"]
[2021/12/30 06:33:44.261 +00:00] [DEBUG] [service.go:181] ["IndexNode stop"] [Address=172.17.0.2:21121]
[2021/12/30 06:33:44.261 +00:00] [DEBUG] [indexnode.go:240] ["Index node stopped."]
[2021/12/30 06:33:44.261 +00:00] [DEBUG] [service.go:195] ["Graceful stop grpc server..."]
[2021/12/30 06:33:44.262 +00:00] [DEBUG] [service.go:132] ["Indexcoord stop"] [Address=172.17.0.2:31000]
2021/12/30 06:33:44 [Builder] close task scheduler
2021/12/30 06:33:44 [Builder] close task scheduler, sched: <nil>
[2021/12/30 06:33:44.262 +00:00] [DEBUG] [service.go:164] ["Datanode stop"] [Address=172.17.0.2:21124]
[2021/12/30 06:33:44.262 +00:00] [DEBUG] [service.go:175] ["Graceful stop grpc server..."]
[2021/12/30 06:33:44.263 +00:00] [DEBUG] [service.go:180] ["Datacoord stop"] [Address=172.17.0.2:13333]
[2021/12/30 06:33:44.263 +00:00] [DEBUG] [service.go:193] ["Graceful stop grpc server..."]
[2021/12/30 06:33:44.264 +00:00] [DEBUG] [service.go:269] ["QueryNode stop"] [Address=172.17.0.2:21123]
[2021/12/30 06:33:44.264 +00:00] [DEBUG] [service.go:281] ["Graceful stop grpc server..."]
[2021/12/30 06:33:44.264 +00:00] [DEBUG] [service.go:282] ["QueryCoord stop"] [Address=172.17.0.2:19531]
[2021/12/30 06:33:44.265 +00:00] [INFO] [query_coord.go:241] ["cancel the loop of QueryCoord"]
[2021/12/30 06:33:44.265 +00:00] [DEBUG] [service.go:294] ["Graceful stop grpc server..."]
[2021/12/30 06:33:44.265 +00:00] [DEBUG] [service.go:339] ["Proxy stop"] [Address=172.17.0.2:19530]
[2021/12/30 06:33:44.265 +00:00] [DEBUG] [service.go:352] ["Graceful stop grpc server..."]
[2021/12/30 06:33:44.265 +00:00] [DEBUG] [service.go:282] ["Rootcoord stop"] [Address=172.17.0.2:53100]
[2021/12/30 06:33:44.265 +00:00] [DEBUG] [service.go:311] ["Rootcoord begin to stop grpc server"]
[2021/12/30 06:33:44.265 +00:00] [DEBUG] [service.go:314] ["Graceful stop grpc server..."]
[2021/12/30 06:33:44.266 +00:00] [DEBUG] [global_rmq.go:107] ["Close Rocksmq!"]
[2021/12/30 06:33:44.266 +00:00] [DEBUG] [rocksmq_retention.go:95] ["Rocksmq retention finish!"]
[2021/12/30 06:33:44.268 +00:00] [INFO] [rocksmq_impl.go:265] ["Successfully close rocksmq"]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x194ff1d]

goroutine 1 [running]:
github.com/milvus-io/milvus/internal/indexcoord.(*TaskScheduler).Close(0x0)
	/data/milvus/internal/indexcoord/task_scheduler.go:305 +0xbd
github.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).Stop(0xc0002e4120, 0x0, 0x0)
	/data/milvus/internal/indexcoord/index_coord.go:295 +0x74
github.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).Stop(0xc000258d20, 0x0, 0x0)
	/data/milvus/internal/distributed/indexcoord/service.go:139 +0x2fd
github.com/milvus-io/milvus/cmd/components.(*IndexCoord).Stop(0xc0000d6d30, 0xc0006bf978, 0x7e39d6)
	/data/milvus/cmd/components/index_coord.go:57 +0x2e
github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc0008afef0, 0x1, 0x0, 0x0)
	/data/milvus/cmd/roles/roles.go:500 +0x5b3
main.main()
	/data/milvus/cmd/main.go:296 +0x635
```

note this:
``` log
2021/12/30 06:33:44 [Builder] close task scheduler
2021/12/30 06:33:44 [Builder] close task scheduler, sched: <nil>
```

**Comment by haorenfsa** [comment]
Looks like the taskscheduler is not inited before shutdown.

A pointer to a struct can call its method even if it's nil.

like this:
```go
package main
type foo struct {
}

func (*foo) bar() {
	print("it works")
}

func main() {
	var p *foo
	p.bar()
}
```

**Comment by DragonDriver** [comment]
@haorenfsa thx for pointing out this. I guess golang is not an OO programming language, so any methods/functions related to a struct will finally be compiled to a single function. The signature of this function will contain the pointer of struct itself and it's valid that we can call a function though it's receiver is nil.

**Comment by DragonDriver** [comment]
/assign @haorenfsa 
Already fixed in #14611 . pls retry this.

**Comment by haorenfsa** [comment]
solved
